### PR TITLE
doge: enable background mining in harness

### PR DESCRIPTION
**Pull request by dev-warrior777**
## Harnesses:

- switch on mining for DGB
- add mining for DOGE

set/export NOMINER="1" to disable

DGB documented to have issues after mining ~100 blocks. Selectively disable as needed.

### CODE REVIEW

**Comment by chappjc** [comment]
I think when we validate 8.22 (once they release final, not rc) we can enable DGB miner.  For now it's broken.

DOGE is good.

**Comment by dev-warrior777** [comment]
What I can do is re-submit a pr with just Doge .. if that works. I have a couple more lines in DGB `harness.sh` that would need adding for 8.x anyway.

I will do that as a new Doge PR if that is ok.

**Comment by chappjc** [comment]
Don't make a new PR pls.  No need for the churn.

File: dex/testing/doge/harness.sh [hunk 1]
```diff
@@ -315,6 +318,18 @@ done
 
 tmux send-keys -t $SESSION:4 "./mine-alpha 2${DONE}" C-m\; ${WAIT}
 
-# Reenable history and attach to the control session.
+################################################################################
+# Setup watch background miner -- if required
+################################################################################
+if [ -z "$NOMINER" ] ; then
+  tmux new-window -t $SESSION:5 -n "miner" $SHELL
```
Comment (chappjc) [inline dex/testing/doge/harness.sh:325]: Please add `$SESSION:5` to the `quit` script up above.

File: dex/testing/doge/harness.sh [hunk 2]
```diff
@@ -273,6 +276,7 @@ tmux send-keys -t $SESSION:0 C-c
 tmux send-keys -t $SESSION:1 C-c
 tmux send-keys -t $SESSION:2 C-c
 tmux send-keys -t $SESSION:3 C-c
+tmux send-keys -t $SESSION:5 C-c
```
Comment (JoeGruffins) [inline dex/testing/doge/harness.sh:279]: Also 4? If I understand correctly, this is just sending ctrl-c to the harnesses to quit whatever they are doing?

Comment (dev-warrior777) [reply dex/testing/doge/harness.sh:279]: I tried both ways .. does not seem to make much difference.

`Window #4` (not 2) is Harness Control for doge. It's the active window as we call `quit` and you could see `^C` on the command line as the only difference. 

Then we kill the session.

Comment (JoeGruffins) [reply dex/testing/doge/harness.sh:279]: Oh, 4 is the harness, then it cannot be doing something other than `./quit` I see

Comment (dev-warrior777) [reply dex/testing/doge/harness.sh:279]: Your question did make me think more .. what if you started a long running process in the harness window/tty programmatically.
In this case you would have to take care of terminating it yourself if you wanted to do that early.
You would just send it a signal!

**Comment by chappjc** [comment]
draft still?

**Comment by dev-warrior777** [comment]
> draft still?

I think it is good to go

**Review by buck54321 — APPROVED** [review]

**Pull request merged**